### PR TITLE
Export manager products

### LINKED ISSUE #45: Export products by Manager Must be export products connected the manager's Shop ONLY!

### PR DESCRIPTION
**Pull request by HMMK99**

### CODE REVIEW

File: xshop/products/admin.py [hunk 1]
```diff
@@ -15,6 +16,14 @@ class ProductAdmin(admin.ModelAdmin):
     search_fields = ("name",)
     ordering = ("-id",)
 
+    def get_queryset(self, request):
+        if request.user.is_superuser:
+            return Product.objects.all()
+        try:
+            return Product.objects.filter(shop=Shop.objects.get(mobile=request.user.mobile))
```
Comment (mmostafa74) [inline xshop/products/admin.py]: try use Product.obejcts.filter(shop__in=request.user.shop)

**Comment by HMMK99** [comment]
Last update I used the mobile to select shop to make sure that the user is the manager of the shop.
as if I select any user connected to the shop then DEC and Cashier can access the data.

File: xshop/products/tests/test_admin.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
+from django.contrib.admin.sites import AdminSite
+from django.test import RequestFactory, TestCase
+from django.core.exceptions import PermissionDenied
+from model_bakery import baker
+
+from ...users.models import GeneralManager, User
```
Comment (mmostafa74) [inline xshop/products/tests/test_admin.py]: Try from xshop.users.models import GeneralManager, User to be more readable

**Review by mmostafa74 — CHANGES_REQUESTED** [review]

File: xshop/products/tests/test_admin.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            list(self.model_admin.get_queryset(self.request_manager).order_by("-id")),
+            list(Product.objects.filter(shop=self.shop).order_by("-id")),
+        )
+
+    def test_manager_no_product_queryset(self):
+        self.assertEqual(
+            list(
+                self.model_admin.get_queryset(self.request_no_product).order_by("-id")
+            ),
+            list(Product.objects.filter(id=0).order_by("-id")),
```
Comment (shahwan42) [inline xshop/products/tests/test_admin.py]: why `id=0`? should you filter by the manager (manager1) of the shop that has no products?
`list(Product.objects.filter(shop=self.manager1.shop).order_by("-id"))`

**Review by shahwan42 — CHANGES_REQUESTED** [review]

**Review by shahwan42 — APPROVED** [review]

**Review by mmostafa74 — APPROVED** [review]

**Pull request merged**